csa-runner-app/src/main/java/com/farao_community/farao/swe_csa/app/dichotomy/DichotomyRunner.java

```diff
@@ -170,14 +173,20 @@ public RaoResult runDichotomy(CsaRequest csaRequest) throws GlskLimitationExcept
                 LOGGER.info("Dichotomy stop criterion reached, CT PT-ES: {}, CT FR-ES: {}", Math.round(index.getBestValidDichotomyStepResult().getCounterTradingValues().getPtEsCt()), Math.round(index.getBestValidDichotomyStepResult().getCounterTradingValues().getFrEsCt()));
                 RaoResult raoResult = index.getBestValidDichotomyStepResult().getRaoResult();
 
-                RaoResultWithCounterTradeRangeActions raoResultWithRangeAction = getRaoResultWithCounterTradeRangeActions(network, crac, index, raoResult);
+                raoResult = updateRaoResultWithVoltageMonitoring(network, crac, raoResult);
+                RaoResultWithCounterTradeRangeActions raoResultWithRangeAction = updateRaoResultWithCounterTradingRAs(network, crac, index, raoResult);
                 fileExporter.saveRaoResultInArtifact(raoResultWithRangeAction, crac, Unit.AMPERE, csaRequest.getBusinessTimestamp());
                 return raoResultWithRangeAction;
             }
         }
     }
 
-    private RaoResultWithCounterTradeRangeActions getRaoResultWithCounterTradeRangeActions(Network network, Crac crac, Index index, RaoResult raoResult) {
+    private static RaoResult updateRaoResultWithVoltageMonitoring(Network network, Crac crac, RaoResult raoResult) {
+        VoltageMonitoring voltageMonitoring = new VoltageMonitoring(crac, network, raoResult);
+        return voltageMonitoring.runAndUpdateRaoResult(LoadFlow.find().getName(), LoadFlowParameters.load(), 1);
```

Comment:
it would be best to use the same LoadFlow name and parameters that are in the RaoParameters object. also, why not parallelize computations?

Reply:
Is there any parameter that i can use to set the number of computation in parallel?

Reply:
this is in the csa runner pod, so i don't know. i guess a new parameter of the app, or automatically detect the number of available cpus